src/Illuminate/Support/Collection.php

```diff
@@ -408,11 +408,15 @@ public function except($keys)
     /**
      * Run a filter over each of the items.
      *
-     * @param  callable|null  $callback
+     * @param  mixed|null  $callback
      * @return static
      */
-    public function filter(callable $callback = null)
```

Comment:
removing a type coersion from a parameter is a breaking change since it changes the behaviour of the function.